### PR TITLE
[SDK]Support cloud eager run submission

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/azure/operations/_flow_operations.py [hunk 1]
```diff
@@ -410,11 +408,7 @@ def _try_resolve_code_for_flow(cls, flow: Flow, ops: OperationOrchestrator, igno
                 return
             if flow._code_uploaded:
                 return
-
-            # TODO(2567532): backend does not fully support generate flow.tools.json from blob storage yet
```
Comment (D-W-) [inline src/promptflow/promptflow/azure/operations/_flow_operations.py:414]: @zhengfeiwang please help check if we still need this logic

Comment (zhengfeiwang) [reply src/promptflow/promptflow/azure/operations/_flow_operations.py:414]: this can be deprecated as it's a workaround during our switch to code-first.

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 2]
```diff
@@ -911,3 +912,84 @@ def test_requirements_in_additional_includes(self, pf: PFClient, randstr: Callab
...
             assert Path(temp, run.name, "snapshot/requirements").exists()
+
+    @pytest.mark.skipif(
+        condition=is_live(),
+        reason="removed requirement.txt to avoid compliance check.",
+    )
+    def test_eager_flow_crud(self, pf: PFClient, randstr: Callable[[str], str]):
+        run = pf.run(
+            flow=f"{EAGER_FLOWS_DIR}/simple_with_req",
+            data=f"{DATAS_DIR}/simple_eager_flow_data.jsonl",
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:923]: Do you think it's necessary to test column mapping functionality also?

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 3]
```diff
@@ -911,3 +912,84 @@ def test_requirements_in_additional_includes(self, pf: PFClient, randstr: Callab
...
+        # # test archive
+        # pf.runs.archive(run=run_id)
+        # run_data = pf.runs._get_run_from_run_history(run_id, original_form=True)[run_meta_data]
+        # assert run_data[hidden] is True
+        #
+        # # test restore
+        # pf.runs.restore(run=run_id)
+        # run_data = pf.runs._get_run_from_run_history(run_id, original_form=True)[run_meta_data]
+        # assert run_data[hidden] is False
+
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:948]: Do we have evaluation test?

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:923]: This PR is only for basic CRUD operations, let me add more complex cases in the future.

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 4]
```diff
@@ -911,3 +912,84 @@ def test_requirements_in_additional_includes(self, pf: PFClient, randstr: Callab
...
+        run = pf.runs.get(run=run_name)
+        # the run status might still be cancel requested, but it should be canceled eventually
+        assert run.status in [RunStatus.CANCELED, RunStatus.CANCEL_REQUESTED]
+
+    @pytest.mark.skipif(
+        condition=is_live(),
+        reason="removed requirement.txt to avoid compliance check.",
+    )
+    @pytest.mark.usefixtures("mock_isinstance_for_mock_datastore")
+    def test_eager_flow_download(self, pf: PFClient, randstr: Callable[[str], str]):
```
[the remark this answers is not in the document]
Comment (D-W-) [reply src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py]: Updated

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/operations/_local_storage_operations.py [hunk 5]
```diff
@@ -217,23 +219,26 @@ def __init__(self, run: Run, stream=False, run_mode=RunMode.Test):
         self._exception_path = self.path / LocalStorageFilenames.EXCEPTION
 
         self._dump_meta_file()
-        if run.flow:
-            try:
-                from promptflow import load_flow
+        self._eager_mode = self._calculate_eager_mode(run)
```
Comment (elliotzh) [inline src/promptflow/promptflow/_sdk/operations/_local_storage_operations.py:222]: NIT: we assume that the type of flow won't be changed via SDK?

**Review by elliotzh — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/operations/_local_storage_operations.py (hunk 5 above)
Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/operations/_local_storage_operations.py:222]: @D-W-  Please resolve this later.

**Pull request merged**